src/points/PointsHistoryBottomSheet.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useMemo } from 'react'
+import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'
+import SectionHead from 'src/components/SectionHead'
+import GorhomBottomSheet from '@gorhom/bottom-sheet'
+import { useDispatch, useSelector } from 'src/redux/hooks'
+import { pointsHistoryStatusSelector, pointsHistorySelector } from 'src/points/selectors'
+import { useTranslation } from 'react-i18next'
+import { typeScale } from 'src/styles/fonts'
+import Logger from 'src/utils/Logger'
+import BottomSheetBase from 'src/components/BottomSheetBase'
+import { Spacing } from 'src/styles/styles'
+import Attention from 'src/icons/Attention'
+import Colors from 'src/styles/colors'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { PointsEvents } from 'src/analytics/Events'
+import { getHistoryStarted } from 'src/points/slice'
+import { ClaimHistory } from 'src/points/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import colors from 'src/styles/colors'
+import { BottomSheetSectionList } from '@gorhom/bottom-sheet'
+
+const TAG = 'Points/PointsHistoryBottomSheet'
+
+interface Props {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+}
+
+function PointsHistoryBottomSheet({ forwardedRef }: Props) {
+  const { t } = useTranslation()
+
+  const dispatch = useDispatch()
+
+  const pointsHistoryStatus = useSelector(pointsHistoryStatusSelector)
+  const pointsHistory = useSelector(pointsHistorySelector)
+
+  const onPressTryAgain = () => {
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_try_again_press)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: false,
+      })
+    )
+  }
+
+  const fetchMoreHistory = () => {
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_fetch_more)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: true,
+      })
+    )
+  }
+
+  const renderError = () => {
+    return (
+      <View testID={'PointsHistoryBottomSheet/ErrorState'} style={styles.errorContainer}>
+        <View style={styles.messageContainer}>
+          <Attention size={48} color={Colors.black} />
+          <Text style={styles.messageTitle}>{t('points.history.error.title')}</Text>
+          <Text style={styles.messageSubtitle}>{t('points.history.error.subtitle')}</Text>
+        </View>
+        <Button
+          testID={'PointsHistoryBottomSheet/TryAgain'}
+          onPress={onPressTryAgain}
+          text={t('points.history.error.tryAgain')}
+          type={BtnTypes.GRAY_WITH_BORDER}
+          size={BtnSizes.FULL}
+          style={{ width: '100%' }}
+        />
+      </View>
+    )
+  }
+
+  const sections = useMemo(() => {
+    if (!pointsHistory.length) {
+      return []
+    }
+    return groupFeedItemsInSections([], pointsHistory, (t: ClaimHistory) => Date.parse(t.createdAt))
```

Comment:
could we simplify this by returning a `timestamp` on the history items?

Reply:
a UNIX timestamp of the `createdAt` time in addition to the `createdAt` field? that would simplify this a bit (the function would be generic on `T extends { timestamp: number }`) though i shy away from it for two reasons:

* we'd be duplicating the "same" data in two different fields, `createdAt` and `timestamp`; having two sources of truth doesn't seem ideal
* we're tightly coupling the API interface with this re-used implementation from the transaction feed, meaning that we're stuck with the name `timestamp` exactly for no great reason (unless we want to apply some transform before this). `timestamp` feels a bit confusing... we already have `createdAt`, so i would assume that `timestamp` might not necessarily _also_ represent creation time

Reply:
the getHistory endpoint is meant to make our lives easy on the front end so i was wondering if we could rename createdAt to timestamp actually...

Reply:
ohh yeah - @jeanregisser was pulling in the direction of making the "created at" time more human readable, but fwiw i actually started with a UNIX timestamp and i think that's probably my preference anyhow

Reply:
In order to avoid another migration, I've made this change _down_ the chain of PRs in `jophish/ret-1032-3`; when I merge that branch into this one, the change will take effect.

Reply:
I have a strong case for keeping ISO 8601 strings: https://github.com/valora-inc/points-functions/pull/46#discussion_r1570453534